networking_generic_switch/devices/corsa_devices/__init__.py

```diff
@@ -207,34 +225,142 @@ def plug_port_to_network(self, port, segmentation_id):
+                br_id = corsavfc.get_bridge_by_segmentation_id(headers, url_switch, segmentation_id)
 
-            try:
-                # unbind the port (probably not necessary)
-                # openflow port was mapped to the physical port with the same port number in plug_port_to_network
-                ofport = self.get_ofport(br_id,port)
-                corsavfc. bridge_detach_tunnel(headers, url_switch, br_id, ofport)
-                LOG.info("needed to delete_port: probably a leaked port from a node that did not completely boot the previous time. ")
-            except Exception as e:
-                LOG.info("Tried to delete_port but it was not there: this is expected." + traceback.format_exc())
-                pass
-
-            # bind the port
-            # physical port is mapped to the openflow port with the same port number
-            ofport = self.get_ofport(br_id,port)
-            corsavfc.bridge_attach_tunnel_passthrough(headers, url_switch, br_id, port_num, ofport, tc = None, descr = None, shaped_rate = None)
+                corsavfc.bridge_attach_tunnel_passthrough(headers, url_switch, br_id, port_num, ofport, tc = None, descr = None, shaped_rate = None)
 
         except Exception as e:
             LOG.error("Failed to plug to network: " + str(traceback.format_exc()))
             raise e
- 
-    def delete_port(self, port, segmentation_id):
+
+
+
+    def __bind_ctag_tunnel(self, segmentation_id, node_vlan, dst_switch_name):
+        LOG.info("adding to vfc_host: __bind_ctag_tunnel")      
+        LOG.info("PRUTH: switch: " + self.config['name'] + ", VFCHost: " + str(segmentation_id) + ", " + str(node_vlan) + ", " + str(dst_switch_name))
+
+        #port_num=port[2:]
+        token = self.config['token']
+        headers = {'Authorization': token}
+
+        logging.basicConfig(format='%(levelname)s:%(message)s', level=logging.DEBUG)
+
+        protocol = 'https://'
+        sw_ip_addr = self.config['switchIP']
+        url_switch = protocol + sw_ip_addr
+        
+        dst_port=None
+        if dst_switch_name in self.config:
+            dst_port=self.config[dst_switch_name]
+
+
+        try:
+            with ngs_lock.PoolLock(self.locker, **self.lock_kwargs):
+                LOG.info("About to dst_port: " + str(dst_port) )
+                br_id = corsavfc.get_bridge_by_segmentation_id(headers, url_switch, segmentation_id)
+                ofport=str(int(str(node_vlan))+10000)
+                LOG.info("ofport: " + str(ofport))
+                LOG.info("br_id: " + str(br_id))
+                corsavfc.bridge_attach_tunnel_ctag_vlan(headers, url_switch, br_id = br_id, ofport = ofport, port = int(dst_port), vlan_id = node_vlan)
+
+        except Exception as e:
+            raise e
```

Comment:
You don't need to catch exceptions if they are being re-raised.

Reply:
Same as the other one... I think this needs to check error codes and raise a new exception.